src/sqlacodegen/generators.py

```diff
@@ -547,6 +553,12 @@ def render_column_type(self, coltype: object) -> str:
             ):
                 del kwargs["astext_type"]
 
+        if isinstance(coltype, DOMAIN):
+            if isinstance(coltype.default, TextClause):
+                kwargs["default"] = render_callable("text", repr(coltype.default.text))
+            if isinstance(coltype.check, TextClause):
+                kwargs["check"] = render_callable("text", repr(coltype.check.text))
```

Comment:
I'm wondering whether this should be a generic thing – turning any `TextClause` instance into `text()` for any argument passed to a callable?

Reply:
To put it differently, is there _any_ case where we would want to render something like `< sqlalchemy.sql.elements.TextClause object at 0x000002173F60A1D0 >`? Because I don't think so, and therefore I would like to convert _all_ kwargs in this manner, thus eliminating the need to special case `DOMAIN`.

Reply:
Don't we always add the import for any column type we render anyway?

Reply:
Oh, my bad, it was `column.type.data_type`. Yeah, we need that one. I wonder if there's a way to get this automatically. But for now I'm fine with special casing `DOMAIN`.

Reply:
So the only change to be made is to generalize `TextClause` -> `text()`.

Reply:
@andrew222651 would you want to take care of it, or would you want me to do that?